artemis-server/src/main/java/org/apache/activemq/artemis/core/remoting/impl/netty/NettyServerConnection.java

```diff
@@ -51,4 +58,55 @@ public void setSNIHostname(String sniHostname) {
    public String getRouter() {
       return router;
    }
+
+   /**
+    * {@return a string representation of the remote address of this connection; if this connection was made via the
+    * proxy protocol then this will be the original address, not the proxy address}
+    */
+   @Override
+   public String getRemoteAddress() {
+      String proxyProtocolSourceAddress = channel.attr(PROXY_PROTOCOL_SOURCE_ADDRESS).get();
+      String proxyProtocolSourcePort = channel.attr(PROXY_PROTOCOL_SOURCE_PORT).get();
+      if (proxyProtocolSourceAddress != null && !proxyProtocolSourceAddress.isEmpty() && proxyProtocolSourcePort != null && !proxyProtocolSourcePort.isEmpty()) {
+         return proxyProtocolSourceAddress + ":" + proxyProtocolSourcePort;
+      } else {
+         return super.getRemoteAddress();
+      }
+   }
+
+   /**
+    * {@return if this connection is made via the proxy protocol then this will be the address of the proxy}
+    */
+   public String getProxyAddress() {
+      String proxyProtocolDestinationAddress = channel.attr(PROXY_PROTOCOL_DESTINATION_ADDRESS).get();
+      String proxyProtocolDestinationPort = channel.attr(PROXY_PROTOCOL_DESTINATION_PORT).get();
+      if (proxyProtocolDestinationAddress != null && !proxyProtocolDestinationAddress.isEmpty() && proxyProtocolDestinationPort != null && !proxyProtocolDestinationPort.isEmpty()) {
+         return proxyProtocolDestinationAddress + ":" + proxyProtocolDestinationPort;
+      } else {
+         return null;
+      }
+   }
+
+   /**
+    * {@return the version of the proxy protocol used to make the connection or null if not applicable}
+    */
+   public String getProxyVersion() {
+      return channel.attr(PROXY_PROTOCOL_VERSION).get() == null ? null : channel.attr(PROXY_PROTOCOL_VERSION).get().toString();
+   }
+
+   public static String getProxyAddress(Connection connection) {
+      if (connection instanceof NettyServerConnection nettyServerConnection) {
```

Comment:
Could a server-specific sub-interface be added, which the server then uses instead of the base Connection, and perhaps avoid either the need of the static method, or at least the need to instanceof here?